src/validate-icon.c

```diff
@@ -224,10 +252,14 @@ rerun_in_sandbox (const char *filename)
 }
 #endif
 
-static gboolean opt_sandbox;
+static gboolean  opt_sandbox;
+static gchar    *opt_path = NULL;
+static gint      opt_fd = -1;
 
 static GOptionEntry entries[] = {
   { "sandbox", 0, 0, G_OPTION_ARG_NONE, &opt_sandbox, "Run in a sandbox", NULL },
+  { "path", 0, 0, G_OPTION_ARG_FILENAME, &opt_path, "Read icon data from given file path, required to be guaranteed non modifiable", "PATH" },
+  { "fd", 0, 0, G_OPTION_ARG_INT, &opt_fd, "Read icon data from given file descriptor, required to be sealed", "FD" },
```

Comment:
This is a bit wrong. Both the (non-sealed) fd and path must not be from an untrusted source. It being "modifiable" isn't an issue and the file xdp passes to it is modifiable for example but xdp creates this file in a way that it's not visible to sandboxed apps (untrusted sources).

Only a sealed fd can be directly from an untrusted source.